hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/prehandle/PreHandleWorkflowImpl.java

```diff
@@ -54,192 +57,211 @@
+                // unit tests, so they use a much shorter configuration, so test execution is faster).
+                final var result = task.future.get(1, TimeUnit.SECONDS);
+                task.platformTx.setMetadata(result);
+            } catch (InterruptedException e) {
+                // The thread should only ever be interrupted during shutdown, so we can just log the error.
+                logger.error("Interrupted while waiting for a transaction to be pre handled.");
+                Thread.currentThread().interrupt();
+                task.platformTx.setMetadata(unknownFailure());
+            } catch (ExecutionException e) {
+                logger.error("Unexpected error while pre handling a transaction!", e);
+                task.platformTx.setMetadata(unknownFailure());
+            } catch (TimeoutException e) {
+                logger.error("Timed out while waiting for a transaction to be pre handled!", e);
+                task.platformTx.setMetadata(unknownFailure());
+            }
+        }
     }
 
-    private PreHandleResult securePreHandle(
-            final HederaState state, final com.swirlds.common.system.transaction.Transaction platformTx)
-            throws PreCheckException {
+    // For each transaction, we will use a background thread to parse the transaction, validate it, lookup the
+    // payer, collect non-payer keys, and warm up the cache. Then, once all the keys have been collected, we will
+    // pass the keys and signatures to the platform for verification.
+    private PreHandleResult preHandleTransaction(
+            @NonNull final AccountID creator,
+            @NonNull final ReadableStoreFactory storeFactory,
+            @NonNull final ReadableAccountStore accountStore,
+            @NonNull final Transaction platformTx) {
+
+        // 1. Parse the Transaction and check the syntax
+        final var txBytes = Bytes.wrap(platformTx.getContents());
+        final TransactionInfo txInfo;
         try {
-            return preHandle(state, platformTx);
-        } catch (Exception ex) {
-            // Some unknown and unexpected failure happened. If this was non-deterministic, I could
-            // end up with an ISS. It is critical that I log whatever happened, because we should
-            // have caught all legitimate failures in another catch block.
-            LOG.error("An unexpected exception was thrown during pre-handle", ex);
-            return createInvalidResult(UNKNOWN);
+            txInfo = transactionChecker.parseAndCheck(txBytes);
+        } catch (PreCheckException preCheck) {
+            // The node SHOULD have verified the transaction before it was submitted to the network.
+            // Since it didn't, it has failed in its due diligence and will be charged accordingly.
+            return nodeDueDiligenceFailure(creator, preCheck.responseCode(), null);
+        } catch (Throwable th) {
```

Comment:
Should this case be logged?

Reply:
I think you are exactly right. This should NEVER happen, and if it does, having it show up in the logs would be super helpful.

Reply:
Better, don't catch Throwable here.  Any exception thrown will be processed as an ExecutionException or, if the suggestion above using CompletableFutures is used, will be passed to the consumer accept method and be handled (and logged) there.